lib/services/resource.go

```diff
@@ -919,3 +926,11 @@ func fieldsFunc(s string, f func(rune) bool) []string {
 
 	return a
 }
+
+var validLabelKey = regexp.MustCompile(`^[a-zA-Z/.0-9_*-]+$`)
```

Comment:
Add a unit test for `IsValidLabelKey` to make sure the regexp is correct

Reply:
I would define the pattern and the `*regexp.Regexp` separately so you can use the same string in `CheckAndSetDefaults` and the JSON schema to avoid drift. Something like:

```go
const labelPattern = `^[a-zA-Z/.0-9_*-]+$`
var validLabelKey = regexp.MustCompile(labelPattern)
```

```go
const MetadataSchema = `{
  "type": "object",
  "additionalProperties": false,
  "default": {},
  "required": ["name"],
  "properties": {
    "name": {"type": "string"},
    "namespace": {"type": "string", "default": "default"},
    "description": {"type": "string"},
    "expires": {"type": "string"},
    "id": {"type": "integer"},
    "labels": {
      "type": "object",
      "additionalProperties": false,
      "patternProperties": {
         "%v":  { "type": "string" }
      }
    }
  }
}`
```

Then you will probably have to update the the functions that return schemas to do something like:

```go
return fmt.Sprintf(V3SchemaTemplate, MetadataSchema, labelPattern, [...])